pip/req/req_install.py

```diff
@@ -778,6 +779,14 @@ def install(self, install_options, global_options=(), root=None):
             self.install_succeeded = True
             return
 
+        # Extend the list of global and install options passed on to
+        # the setup.py call with the ones from the requirements file.
+        # Options specified in requirements file override those
+        # specified on the command line, since the last option given
+        # to setup.py is the one that is used.
```

Comment:
Good memory - that was 2 years ago :)

It's weird indeed and I think it should be mentioned somewhere that using this feature will void your warranty. There is nothing to stop people from having something like the following in their requirements:

```
six==1.9.0 --global-options="--dry-run"
```